File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var hash = {};

    for (let i = 0; i < nums.length; i++)
    {
        const num = nums[i];
        const diff = target - num;

        if (hash[diff] !== undefined)
        {
            return [hash[diff], i];
        }
        hash[num] = i;
    }
    return []
};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int nL = nums.Length;
        Dictionary<int, int> dict = new Dictionary<int, int>(nL);
       
       for (int i=0; i<nL; i++)
       {
           int diff = target - nums[i];
           if(dict.ContainsKey(diff))
               return new int[] {i, dict[diff]};
            dict[nums[i]]= i;
       }
       return new int[] {0,0};
    }
}